samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -203,6 +204,15 @@ class TaskInstance(
     trace("Checkpointing offsets for taskName: %s" format taskName)
 
     offsetManager.writeCheckpoint(taskName, checkpoint)
+
+    if (checkpoint != null) {
+      checkpoint.getOffsets.asScala
+        .filter { case (ssp, _) => intermediateStreams.contains(ssp.getStream) } // Only delete data of intermediate streams
+        .groupBy { case (ssp, _) => ssp.getSystem }
+        .foreach { case (systemName: String, offsets: Map[SystemStreamPartition, String]) =>
+          systemAdmins.getSystemAdmin(systemName).deleteMessages(offsets.asJava)
```

Comment:
IIUC, because of the default implementation, this will throw for all systems except Kafka. That doesn't seem correct.

Reply:
Good catch @jmakes. I have updated the patch to fix it by not throwing exception in the default implementation.

Reply:
Great find @jmakes! This was missed in my earlier review!